File: 프로그래머스/1/140108. 문자열 나누기/문자열 나누기.py
def solution(s):
    answer = 0
    
    target = s[0]
    same_string_count = 0
    different_string_count = 0
    
    for i in range(len(s) - 1):
        if target == s[i]:
            same_string_count += 1
        elif target != s[i]:
            different_string_count += 1
        
        if same_string_count == different_string_count:
            target = s[i + 1]
            same_string_count = 0
            different_string_count = 0
            answer += 1
    
    return answer + 1

File: 프로그래머스/1/140108. 문자열 나누기/문자열 나누기.js
function solution(s) {
    let answer = 0;
    
    let target = "";
    let targetCount = 0;
    let compareCount = 0;
    for (const item of s) {
        if (target) {
            if (item === target) {
                targetCount++;
            } else {
                compareCount++;
            };
            if (targetCount === compareCount) {
                answer += 1;
                target = "";
                targetCount = 0;
                compareCount = 0;
            };
        } else {
            target = item;
            targetCount++;
        }
    }
        
    if (targetCount || compareCount) answer++;
    
    return answer;
}